### PR TITLE
[BEAM-10082] Remove dependency on java.xml.ws and remove deprecated MetricName methods

### PR DESCRIPTION
**Pull request by tysonjh**
Java 11 removed some Java EE modules. The dependency in MetricsHttpSink
is for HttpException. Replace this instead with an IOException. Also clean up
a couple deprecated methods, get and namespace, in the process.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Samza | Spark
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by tysonjh** [comment]
R: @echauchot 

Are you able to review this PR please?

**Comment by apilloud** [comment]
retest this please

**Comment by echauchot** [comment]
@tysonjh thanks for your work ! Reviewing right now

File: runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java [hunk 1]
```diff
@@ -50,6 +49,12 @@ public MetricsHttpSink(MetricsOptions pipelineOptions) {
     this.urlString = pipelineOptions.getMetricsHttpSinkUrl();
   }
 
+  /**
+   * Writes the metricQueryResults via HTTP POST to metrics sink endpoint.
+   *
+   * @param metricQueryResults query results to write.
+   * @throws Exception throws IOException for non-200 response from endpoint.
+   */
```
Comment (echauchot) [inline runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java:57]: thanks for adding this missing javadoc

File: runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java [hunk 2]
```diff
@@ -69,7 +74,7 @@ public void writeMetrics(MetricQueryResults metricQueryResults) throws Exception
     }
     int responseCode = connection.getResponseCode();
     if (responseCode != 200) {
-      throw new HTTPException(responseCode);
+      throw new IOException("Expected OK 200 response but received: " + responseCode);
```
Comment (echauchot) [inline runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java:77]: Agreed as there is no suitable http exception in java.net regular package. But please change the message to add more context "Exception while writing metrics to MetricsHttpSink : expected HTTP 200 OK response but received "

**Comment by tysonjh** [comment]
Thanks for the review @echauchot, it is ready for another look.

File: runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java [hunk 3]
```diff
@@ -86,8 +93,8 @@ public MetricNameSerializer(Class<MetricName> t) {
     public void serialize(MetricName value, JsonGenerator gen, SerializerProvider provider)
         throws IOException {
       gen.writeStartObject();
-      gen.writeObjectField("name", value.name());
-      gen.writeObjectField("namespace", value.namespace());
```
[the remark this answers is not in the document]
Comment (echauchot) [reply runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java:90]: yes I replaced them with get* versions to simplify serialization code but they were re-introduced to be compatible with dataflow hosted engine which uses .name and .namespace which I could not know of because it is not opensource. Anyway good catch !  Dataflow is supposed to have migrated to .get* versions now, can you check ? If so we can remove the deprecated versions from the api.

File: runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java [hunk 4]
```diff
@@ -74,7 +74,9 @@ public void writeMetrics(MetricQueryResults metricQueryResults) throws Exception
     }
     int responseCode = connection.getResponseCode();
     if (responseCode != 200) {
-      throw new IOException("Expected OK 200 response but received: " + responseCode);
+      throw new IOException(
+          "Expected HTTP 200 OK response while writing metrics to MetricsSinkHttp but received "
```
[the remark this answers is not in the document]
Comment (tysonjh) [reply runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java]: Oops! Done.

File: runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tysonjh) [reply runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java:90]: Got it, I was just doing some code cleaning as I went along. I've verified that removing the deprecated methods doesn't break anything and added that to this PR. I'll update the commit message as well.

**Comment by echauchot** [comment]
LGTM! Thanks for the changes, that's perfect. Last thing to do is clean review comments: can you squash the first 2 commits and leave only the comment about dependency. The 3rd commit needs to be appart IMHO because it changes the (deprecated) API. Though, reword it to: "remove deprecated MetricName.name() and MetricName.namespace() methods and fix a typo". 
regarding build failure, it's an unrelated timeout, a new build will be triggered when you push.
Thanks

**Comment by tysonjh** [comment]
> LGTM! Thanks for the changes, that's perfect. Last thing to do is clean review comments: can you squash the first 2 commits and leave only the comment about dependency. The 3rd commit needs to be appart IMHO because it changes the (deprecated) API. Though, reword it to: "remove deprecated MetricName.name() and MetricName.namespace() methods and fix a typo".
> regarding build failure, it's an unrelated timeout, a new build will be triggered when you push.
> Thanks

Done.

**Comment by echauchot** [comment]
Perfect ! merging
